### PR TITLE
UAF-2778 -  Several PO Subtypes Fail To Load In UI, Throws STE

### PR DESCRIPTION
**Pull request by sskinner**
This PR moves two fields from multiple PO subclasses in the AZ package, to a common base class in the Kuali package.This allows all PO subclasses to gain the needed fields at once, which is necessary for JSPs, searches, DD, and other mechanisms.The remainder of changes here are to effect this in Spring and OJB. This trades off having 30 lines of UA code being in one Kuali class, in order to remove 1000+ lines of redundant code/config in the UA package. Also noteworthy, is that the nature of this being a core object, and being in a hierarchy and not service, preclude us from being able to do so without either adding massive amounts of redundant extension code/config, or alternatively do what's done here and having to slightly change a kuali package. Please see the corresponding jira for more rationale, and a thorough list of file changes.

### CODE REVIEW

File: kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java [hunk 1]
```diff
@@ -41,6 +32,15 @@
 import org.kuali.kfs.sys.KFSPropertyConstants;
 import org.kuali.rice.core.api.util.type.KualiDecimal;
 
+import java.util.ArrayList;
```
Comment (ghost) [inline kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java:35]: Just wondering why this group of imports was moved?

File: kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java [hunk 2]
```diff
@@ -201,9 +201,9 @@
             reqAppDocStatusMap.put(NODE_ACCOUNT, APPDOC_DAPRVD_FISCAL);
             reqAppDocStatusMap.put(NODE_ORG_REVIEW, APPDOC_DAPRVD_CHART);
             reqAppDocStatusMap.put(NODE_COMMODITY_CODE_REVIEW, APPDOC_DAPRVD_COMMODITY_CODE);
-            reqAppDocStatusMap.put(NODE_SEPARATION_OF_DUTIES, APPDOC_DAPRVD_SEP_OF_DUTY);
+            reqAppDocStatusMap.put(NODE_SEPARATION_OF_DUTIES, APPDOC_DAPRVD_SEP_OF_DUTY);
             reqAppDocStatusMap.put(NODE_SUB_FUND, APPDOC_DAPRVD_SUB_FUND);
-            reqAppDocStatusMap.put(NODE_OBJECT_SUB_TYPE_CODE, APPDOC_DAPRVD_OBJECT_SUB_TYPE_CODE);
```
Comment (ghost) [inline kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java:206]: There seems to be several lines showing as changed in this file with no apparent change. Editor issues?

File: kfs-purap/src/main/java/org/kuali/kfs/module/purap/document/PurchasingDocumentBase.java [hunk 3]
```diff
@@ -72,12 +69,19 @@
 import org.kuali.rice.location.api.country.Country;
 import org.kuali.rice.location.api.country.CountryService;
 
+import java.sql.Date;
```
Comment (ghost) [inline kfs-purap/src/main/java/org/kuali/kfs/module/purap/document/PurchasingDocumentBase.java:72]: Again, imports being moved for no apparent reason.

File: kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java (hunk 1 above)
Comment (sskinner) [inline kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java:35]: It's an algorithm in IntelliJ for organizing imports. I would turn it off, but then we lose the "delete unused" functionality, and also organizing for when imports really are a mess. In the end, it doesn't matter to the compiler, but nonetheless frustrating for viewing by humans.

File: kfs-purap/src/main/java/org/kuali/kfs/module/purap/document/PurchasingDocumentBase.java (hunk 3 above)
Comment (sskinner) [inline kfs-purap/src/main/java/org/kuali/kfs/module/purap/document/PurchasingDocumentBase.java:72]: I think it separates the Java API imports from the non-API, and then applies alphabetical sort within the those sub-categories. Didn't recognize it till now, so maybe I'll have half a chance at finding it in the settings to disable it.

File: kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java (hunk 2 above)
Comment (sskinner) [inline kfs-purap/src/main/java/org/kuali/kfs/module/purap/PurapConstants.java:206]: Yes, and the frustrating thing is I did not intentionally modify these lines, it was some automated IntelliJ action. I'm not even sure what the actual difference that was introduced -- presumably whitespace, even though I have "replace tabs/newlines" turned off.

**Comment by ghost** [comment]
Looks good!

**Pull request merged**